src/components/SortableTable/SortableTable.tsx

```diff
@@ -0,0 +1,270 @@
+import classnames from 'classnames';
+import uniqueId from 'lodash.uniqueid';
+import type { Key, MouseEvent, ReactNode } from 'react';
+import React from 'react';
+import Button from '../Button';
+import Icon from '../Icon';
+import Label from '../Label';
+import type { HeaderProps } from './Header';
+import Header from './Header';
+import type { TableProps } from '../Table';
+import Table from '../Table';
+
+type HorizontalAlignment = 'left' | 'center' | 'right';
+
+export interface SortableColumn<T>
+  extends Omit<HeaderProps, 'children' | 'onSort'> {
+  align?: HorizontalAlignment;
+  cell?: (row: T, expanded?: boolean) => ReactNode;
+  footer?: ReactNode;
+  header?: ReactNode;
+  hidden?: boolean;
+  key?: Key;
+  onSort?: (ascending: boolean) => void;
+  sortable?: boolean;
+  width?: string;
+}
+
+export type SortableTableRow<T> = T & { key: Key | undefined };
+type RowClassNameFunction<T> = (row: T) => ReactNode | undefined;
+type RowExpandedFunction<T> = (row: T) => ReactNode | false;
+type RowOnClickFunction<T> = (row: T, evt: MouseEvent) => void;
+type RowSelectedFunction<T> = (row: T) => boolean;
+
+export interface SortableTableProps<T> extends Omit<TableProps, 'children'> {
+  allSelected?: boolean;
+  columns: SortableColumn<T>[];
+  expandableColumn?: Partial<SortableColumn<T>>;
+  footer?: ReactNode;
+  key?: string;
+  onExpand?: (row: T) => void;
+  onSelect?: (row: T, selected?: boolean) => void;
+  onSelectAll?: (selected: boolean) => void;
+  /* eslint-disable-next-line no-use-before-define */
+  renderRow?: typeof defaultRenderRow;
+  rowClassName?: RowClassNameFunction<T>;
+  rowExpanded?: RowExpandedFunction<T>;
+  rowOnClick?: RowOnClickFunction<T>;
+  rows?: SortableTableRow<T>[];
+  rowSelected?: RowSelectedFunction<T>;
+  truncate?: boolean;
+}
+
+function generateColumnClassName<T>(
+  column: SortableColumn<T>,
+  truncate = false
+) {
+  return classnames(
+    truncate && 'text-truncate',
+    column.align && `text-${column.align}`,
+    column.className
+  );
+}
+
+function defaultRenderRow<T>(
+  row: SortableTableRow<T>,
+  columns: SortableColumn<T>[],
+  rowClassName?: RowClassNameFunction<T>,
+  rowExpanded?: RowExpandedFunction<T>,
+  rowOnClick?: RowOnClickFunction<T>,
+  truncate?: boolean,
+  rowSelected?: RowSelectedFunction<T>
```

Comment:
Got a few. Unfortunately, I'd have to rearrange the order of the args to make it the way we'd want completely. That would break the interface.
Since this might be on the chopping block in favor of a composable element, I'll keep it simple.